src/history/__tests__/history.test.tsx

```diff
@@ -0,0 +1,76 @@
+import Enzyme, { mount, shallow } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+import History, { HistoryChart } from 'history/history'
+import React from 'react'
+import renderer from 'react-test-renderer'
+
+import {
+  mockEmptyHistoryContainerProps,
+  mockHistoryChartProps,
+  mockHistoryContainerProps
+} from './__mock__/mockData'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+const createNodeMock = () => {
+  const doc = document.implementation.createHTMLDocument()
+  return { parentElement: doc.body }
+}
+
+describe('history component', () => {
+  it('should render history chart correctly', () => {
+    const tree = renderer
+      .create(<HistoryChart {...mockHistoryChartProps} />, { createNodeMock })
+      .toJSON()
+
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('should render correctly', () => {
+    const tree = renderer
+      .create(<History {...mockHistoryContainerProps} />, { createNodeMock })
+      .toJSON()
+
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('should render correctly with no historical data', () => {
+    const wrapper = shallow(<History {...mockEmptyHistoryContainerProps} />)
+
+    expect(wrapper.getElement()).toBeNull()
+  })
+
+  it('should render filter buttons correctly', () => {
+    const wrapper = shallow(<History {...mockHistoryContainerProps} />)
+
+    expect(wrapper.find('FilterButton[type="date"]').length).toBe(6)
+    expect(wrapper.find('FilterButton[type="price"]').length).toBe(4)
+  })
+
+  describe("should change filter button's selected attribute when filter is changed", () => {
+    const wrapper = shallow(<History {...mockHistoryContainerProps} />)
```

Comment:
This would be better handled in a `beforeEach` block (or just called in each test). It affects the execution order of jest - if this call was to error then that would happen while jest is initialising, instead of when running this test unit.

Avoiding duplication is less important in test code (it's more important to avoid coupling all your tests together so that you can make changes without invalidating the entire test suite)